### DISCUSSION TITLE
Detect if an Edge is outside or inside a Mesh

### DISCUSSION THREAD

**Discussion by mobereng** [libigl/libigl#1763] [Q&A] [answered]
Hi guys,

Currently I'm trying to detect if a line / segment  / edge, defined by 2 points v1, v2 is completely inside the mesh or any part of the line is outside of it, my goal is to return only a single boolean, and my current mesh is a triangular one (V, F)

**Comment by jdumas** [comment, marked as answer]
You could raytrace to see if your segment intersects any triangle, and then use winding number to determine if you are inside/outside. Raytracing won't be very robust though, but that's a start.

**Reply by mobereng** [reply]
Thanks ! got it, let me try it
